extendr-api/src/wrapper/rstr.rs

```diff
@@ -61,10 +61,20 @@ impl From<&str> for Rstr {
     }
 }
 
+impl From<Option<String>> for Rstr {
+    fn from(value: Option<String>) -> Self {
+        if let Some(string) = value {
+            Self::from(string)
+        } else {
+            Self { robj: na_string() }
+        }
+    }
+}
+
```

Comment:
I see literally no downsides with this! Beautiful! Just out of curiosity since I still dont really get the use of `if let` what is the use of it here instead of 

```rust
match value {
    Some(v) => Self::from(v),
    None => na_string()
}
```

Reply:
`match` looks succinct here. I should have written it like this. But `match` looks innocent, but it is very flexible. `if let` is a match on "patterns" only. While you could have a match arm include a test of value.
So to me, when I see `if let` I see a constrained construction, and it is sort of easier for me to read.
I don't know what to say really. `match` 

See this for an example of "dangerous" uses of `match`: https://github.com/rust-lang/rust/issues/37612